extensions/EXT/EGL_EXT_device_query_vendor.txt

```diff
@@ -0,0 +1,77 @@
+Name
+
+    EXT_device_query_vendor
+
+Name Strings
+
+    EGL_EXT_device_query_vendor
+
+Contributors
+
+    Eric Engestrom (eric 'at' engestrom.ch)
+
+Contact
+
+    Eric Engestrom (eric 'at' engestrom.ch)
+
+Status
+
+    Draft
+
+Version
+
+    Version 1, Jun 06, 2020
+
+Number
+
+    EGL Extension #140
+
+Extension Type
+
+    EGL device extension
+
+Dependencies
+
+    This extension is written against the language of EGL 1.5 as
+    modified by EGL_EXT_device_query.
+
+    EGL_EXT_device_query is required.
+
+Overview
+
+    This extension adds a simple way to query the PCI vendor ID and device ID.
+
+IP Status
+
+    No known claims.
+
+New Types
+
+    None.
+
+New Procedures and Functions
+
+    None.
+
+New Tokens
+
+    None.
+
+Changes to section 3.2 (Devices)
+
+    Replace the paragraph immediately following the prototype for
+    eglQueryDeviceAttribEXT:
+
+    <attribute> may be either EGL_QUERY_VENDOR_ID_MESA or EGL_QUERY_DEVICE_ID_MESA.
+    On success, EGL_TRUE is returned, and the EGL device's PCI vendor ID or device ID
+    is returned in <value>. If the EGL device is not on a PCI bus, or is not
+    a hardware device at all, EGL_FALSE is returned and <value> is unchanged.
```

Comment:
Should we throw EGL_BAD_ATTRIBUTE, too?  Doesn't EGL generally always throw an error on false returns?  (genuine question, I'm unsure)

Other than that, this sounds great to me.

Reply:
Every EGLDeviceEXT has its own extension string. If it's not a PCI device, then it should just not advertise this extension at all.

Reply:
I like that solution.

Reply:
In mesa, we advertise the same client extensions for all the drivers/devices. I haven't looked at how to change that, but I think I'd prefer if we allowed to advertise this extension on non-PCI devices and have it return `EGL_FALSE`

>  Doesn't EGL generally always throw an error on false returns? (genuine question, I'm unsure)

I can't think of anything off the top of my head (too tired), but I think there are other cases where "false" is not an api error and just a "nope, can't do this" error

Reply:
OK, so I was misunderstanding device extensions quite a bit, thanks Kyle for pointing things out to me.

I have looked into the way we do things in Mesa and, while not very memory efficient (we'd waste a few dozen bytes of static memory per device), it's trivial to only expose the extension when the device is on a PCI bus, so let's do that :+1:

I've updated the spec text, let me know if this is ok?

Reply:
Marking as resolved, since there isn't any possible error anymore